File: AggieAutoSchedulerAngular/ClientApp/src/app/classes/exam.ts
export class Exam {
  professor: string;
  location: string;
  startTime: Date;
  endTime: Date;
  constructor(professor?: string, location?: string, startTime?: Date, endTime?: Date) {
    this.professor = professor;
    this.location = location;
    this.startTime = startTime;
    this.endTime = endTime;
  }
}

File: AAS.ScheduleMaker/Classes/Exam.cs
using System;
namespace AAS.ScheduleMaker
{
    public class Exam
    {
        public Exam(AAS.DB.Exam exam)
        {
            Professor = exam.Professor;
            Location = exam.Location;
            StartTime = exam.StartTime;
            EndTime = exam.EndTime;
        }

        public string Professor { get; set; }
        public string Location { get; set; }

        public DateTime StartTime { get; set; }

        public DateTime EndTime { get; set; }

        public bool OverlapsWith(Exam other)
        {
            return EndTime >= other.StartTime && other.EndTime >= StartTime; 
        }
    }
}
